### PR TITLE
Balder/separate hot cold path tomake fast path faster

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe and @geirst PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp [hunk 1]
```diff
@@ -56,56 +55,56 @@ MultiBitVector<Update>::MultiBitVector(size_t reserved)
       _accel(IAccelrated::getAccelerator()),
       _lastWords()
 {
-    static_assert(sizeof(_lastWords) == 64, "Lastwords should have 64 byte size");
-    static_assert(NumWordsInBatch == 8, "Batch size should be 8 words.");
+    static_assert(sizeof(_lastWords) == 256, "Lastwords should have 64 byte size");
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp:58]: Message doesn't match the assert.

File: searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp [hunk 2]
```diff
@@ -56,56 +55,56 @@ MultiBitVector<Update>::MultiBitVector(size_t reserved)
...
+    static_assert(NumWordsInBatch == 32, "Batch size should be 8 words.");
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp:59]: Message doesn't match the assert.

File: vespalib/src/vespa/vespalib/hwaccelrated/iaccelrated.h [hunk 3]
```diff
@@ -32,9 +32,9 @@ class IAccelrated
     virtual double squaredEuclideanDistance(const float * a, const float * b, size_t sz) const noexcept = 0;
     virtual double squaredEuclideanDistance(const double * a, const double * b, size_t sz) const noexcept = 0;
     // AND 64 bytes from multiple, optionally inverted sources
-    virtual void and64(size_t offset, const std::vector<std::pair<const void *, bool>> &src, void *dest) const noexcept = 0;
+    virtual void and256(size_t offset, const std::vector<std::pair<const void *, bool>> &src, void *dest) const noexcept = 0;
```
Comment (geirst) [inline vespalib/src/vespa/vespalib/hwaccelrated/iaccelrated.h:35]: Function comment should be updated.

File: vespalib/src/vespa/vespalib/hwaccelrated/iaccelrated.h [hunk 4]
```diff
@@ -32,9 +32,9 @@ class IAccelrated
...
     // OR 64 bytes from multiple, optionally inverted sources
-    virtual void or64(size_t offset, const std::vector<std::pair<const void *, bool>> &src, void *dest) const noexcept = 0;
+    virtual void or256(size_t offset, const std::vector<std::pair<const void *, bool>> &src, void *dest) const noexcept = 0;
```
Comment (geirst) [inline vespalib/src/vespa/vespalib/hwaccelrated/iaccelrated.h:37]: Function comment should be updated.

**Review by geirst — DISMISSED** [review]
👍

File: searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp]: Ficed

File: vespalib/src/vespa/vespalib/hwaccelrated/iaccelrated.h (hunk 3 above)
Comment (baldersheim) [reply vespalib/src/vespa/vespalib/hwaccelrated/iaccelrated.h:35]: Fixed

File: searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp]: Fixed

File: vespalib/src/vespa/vespalib/hwaccelrated/iaccelrated.h (hunk 4 above)
Comment (baldersheim) [reply vespalib/src/vespa/vespalib/hwaccelrated/iaccelrated.h:37]: Fixed

**Pull request merged**